libcodechecker/libhandlers/analyze.py

```diff
@@ -177,6 +184,58 @@ def add_arguments_to_parser(parser):
                                help="File containing argument which will be "
                                     "forwarded verbatim for Clang-Tidy.")
 
+    if is_ctu_capable():
+        ctu_opts = parser.add_argument_group('cross translation unit analysis')
+        ctu_mutex_group = ctu_opts.add_mutually_exclusive_group()
+        ctu_mutex_group.add_argument('--ctu',
+                                     action='store_const',
+                                     const=[True, True],
+                                     dest='ctu_phases',
+                                     default=argparse.SUPPRESS,
+                                     help="Perform cross translation unit "
+                                          "(CTU) analysis (both collect and "
+                                          "analyze phases) using default "
+                                          "<ctu-dir> for temporary output. "
+                                          "At the end of the analysis, the "
+                                          "temporary directory is removed.")
```

Comment:
The original concept was is to merge this code with as little changes as possible to not diverge too much from the ctu code in scan-build. Options and everything also fits the scan-build implementation.

The tools create auxiliary data for CTU. This means that they do not really help debugging a clang crash as this information is kept elsewhere anyway. I would say, this data is not particularly useful unless we want to debug these tools (clang-func-mapping) themselves. Still the real and only reason removing the temporary folder is user's convenience, so nothing will happen if we don't do it.